stream_utils.py

```diff
@@ -4,6 +4,45 @@
 from amaranth.utils import ceil_log2
 
 
+def tree_and(m: Module, inputs: list[Signal]) -> Signal:
+    if len(inputs) == 1:
+        return inputs[0]
+    elif len(inputs) == 2:
+        result = Signal(1)
+        m.d.comb += result.eq(inputs[0] & inputs[1])
+        return result
+    else:
+        l = len(inputs)
+        left = tree_and(m, inputs[:l//2])
+        right = tree_and(m, inputs[l//2:])
+        return tree_and(m, [left, right])
```

Comment:
I wonder if the Amaranth reduction ops end up synthesizing to the same thing?

https://amaranth-lang.org/docs/amaranth/latest/guide.html#reduction-operators